apps/src/code-studio/pd/workshop_enrollment/enrollmentCancelButton.jsx

```diff
@@ -0,0 +1,66 @@
+import React, {PropTypes} from 'react';
+import {Button} from 'react-bootstrap';
+import ConfirmationDialog from '../components/confirmation_dialog';
+
+export default class EnrollmentCancelButton extends React.Component {
+  static propTypes = {
+    enrollmentCode: PropTypes.string.isRequired,
+    workshopFriendlyName: PropTypes.string.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      showConfirmation: false,
+      canceled: false
+    };
+  }
+
+  componentWillUnmount() {
+    if (this.cancelEnrollmentRequest) {
+      this.cancelEnrollmentRequest.abort();
+    }
+  }
+
+  handleCancelClick = () => {
+    this.setState({showConfirmation: true});
+  };
+
+  handleCancelConfirmed = () => {
+    this.cancelEnrollmentRequest = $.ajax({
+      method: 'DELETE',
+      url: `/api/v1/pd/enrollments/${this.props.enrollmentCode}`,
+      dataType: "json"
+    }).done(() => {
+      window.location.reload(true);
+      // this.setState({canceled: true});
+    });
+  };
+
+  handleCancelAborted = () => {
+    this.setState({showConfirmation: false});
+  };
+
+  render() {
+    return (
+      <div>
+        <p>Click below to cancel your registration in the {this.props.workshopFriendlyName}.</p>
```

Comment:
Does this include the date?

Reply:
Yes. See https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/pd/workshop.rb#L257